.github/workflows/tests.yml

```diff
@@ -0,0 +1,28 @@
+name: Run unit tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    strategy:
+      matrix:
+        python-version: ["3.11", "3.12", "3.13"]
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          pip install -U pip
+          pip install -r requirements.txt
+      - name: Run all tests
+        run: |
+          pytest .
```

Comment:
🤔 Suggestion but no blocker at all - I find it's nice to have `--verbose` test outputs in CI but I also think inspections as-is are alright! 🚀